src/main/commandServer/__tests__/settingsValidator.spec.ts

```diff
@@ -135,9 +136,9 @@ describe(SettingsValidator, () => {
       const [needToUpdate, errors] = subject.validateSettings(cfg, {
         kubernetes: {
           version:         '1.1.1',
-          containerEngine: '1.1.2',
-          enabled:         1,
-          options:         { flannel: 1 },
+          containerEngine: '1.1.2' as settings.ContainerEngine,
```

Comment:
I gather all this typing is being done to deal with arbitrary user input? A comment on the type strategy (including `as unknown as T` for some type `T` would be helpful as well.

Reply:
Yeah, basically we're going against TypeScript here to generate invalid data to test against. (Note that none of the mess is outside test files.)